### ISSUE TITLE
Some confusion about real overlap calculation

### ISSUE THREAD

**Issue by ZYJ0409** [PRBonn/OverlapNet#30]
Dear author
    I had some problems running your code with the overlap truth calculation. In the 00 sequence, frame 0 and frame 4449 are the same region, but their true overlap is only 0.14.
<img width="605" alt="微信图片_20220411160408" src="https://user-images.githubusercontent.com/78292641/162692091-3c46e83b-0a7f-4eb0-b323-7879de885269.png">
<img width="238" alt="微信图片_20220411160444" src="https://user-images.githubusercontent.com/78292641/162692291-786e3686-fc81-4c1a-941c-b6904ccd2b7c.png">
I look forward to your reply. Thank you

**Comment by Chen-Xieyuanli** [comment]
Hey @ZYJ0409, thanks for using our code!

In my case, it's more than 80. Which code are you using? You may double check the poses you are using.

![Screenshot from 2022-04-11 11-40-22](https://user-images.githubusercontent.com/17588237/162711670-f49fcdee-7539-4a8c-8db2-0a8b861905b9.png)

**Comment by ZYJ0409** [comment]
Dear author
Thank you for your reply. I refer to '**com_overlap_yaw ()**' in your code. I enter '**current_POSES**' in frame 0 (white) and '**reference_IDx**' in frame 4449 (blue). This is the pose of my two frames, but the overlap area of my output is only 0.14. How do you solve the error caused by the scale drift of kitti dataset?
<img width="1002" alt="微信图片_白色" src="https://user-images.githubusercontent.com/78292641/162777846-40678b97-1ca1-492e-a2a0-7d2dab5398e8.png">
<img width="1001" alt="微信图片_蓝色" src="https://user-images.githubusercontent.com/78292641/162777866-1cf24182-313c-459e-b197-d1635d273877.png">
<img width="459" alt="微信图片_当前帧" src="https://user-images.githubusercontent.com/78292641/162777927-8780fce6-3c4c-488f-b0ad-535c127b1606.png">
<img width="633" alt="微信图片_检测帧" src="https://user-images.githubusercontent.com/78292641/162777955-975f9a75-8998-434d-91ae-704e9d5a3bf5.png">
Looking forward to your reply.

**Comment by Chen-Xieyuanli** [comment]
I think the error comes from the poses. Are you using the ground truth poses? They are in the camera coordinate systems and you need to first convert them into the LiDAR coordinate system.

**Comment by ZYJ0409** [comment]
Dear author
I'm sorry that's my problem. I didn't express myself clearly. This is indeed the camera pose, but your code has been converted in "demo4_gen_gt_files", I have taken this into account, but the result is still very low
<img width="519" alt="微信图片_convert" src="https://user-images.githubusercontent.com/78292641/162856495-a32bdd9a-a532-4ef7-ad0b-1ad4a88f1848.png">

**Comment by ZYJ0409** [comment]
This is our transformation method. If possible, could you please provide your algorithm or source code? Thank you very much！
![微信图片_20220412092437](https://user-images.githubusercontent.com/78292641/162860012-6eac9646-17e9-47b7-aa22-97a193426f67.jpg)

**Comment by Chen-Xieyuanli** [comment]
How did you generate the poses? They seem not accurate and not the ground truth poses.

**Comment by ZYJ0409** [comment]
Dear author
    I downloaded them from the Internet， Can you provide a link to download the ground truth poses? (the best is '**baidu cloud**’)，thank you very much, Your reply was very helpful to us!

**Comment by Chen-Xieyuanli** [comment]
You could try these two together with our code.
[calib.txt](https://github.com/PRBonn/OverlapNet/files/8470921/calib.txt)
[poses.txt](https://github.com/PRBonn/OverlapNet/files/8470922/poses.txt)

**Comment by ZYJ0409** [comment]
Thank you very much for your patient help. There is something wrong with our data. Wish you a happy life and smooth work!

**Comment by Chen-Xieyuanli** [comment]
I'm glad that the problem is solved! I will then close this issue.

I'll appreciate it if you could star our repo to let more people see it if you find our code and answers helpful.

**Comment by BlingLee** [comment]
![Uploading 2024-08-28_20-36.png…]()
> 非常感谢您的耐心帮助。我们的数据有问题。祝您生活愉快，工作顺利！
Hello. How are you? I also tried to reproduce overlapnet, but I met some problems, can I ask you for advice?

In demo4, we need to enter poses.txt and calib.txt as well as scan. May I ask what processing is required for this poses. I download the poses file for the kitti dataset directly from the Internet, and the authors of the paper also provide the poses file for the 07 sequence, but for the 07 sequence, the file that I download is not the same as the Poses file provided by the authors. Okay? poses.txt is converted in demo4.
![2024-08-28_20-36](https://github.com/user-attachments/assets/75267e7a-34d2-40cc-8010-59b68d1e3c70)

This is the 07 sequence poses.txt that I downloaded. Looking forward to your reply
